### ISSUE TITLE
Test failure: CheckInstrBytePattern(base[offset ] & 0xFD, 0x81, base[offset]) && CheckInstrBytePattern(base[offset+1] & 0xC0, 0xC0, base[offset+1])

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#45090]
failed in job: [runtime-coreclr gcstress0x3-gcstress0xc 20201122.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=895872&view=ms.vss-test-web.build-test-results-tab&runId=28629800&resultId=100157&paneView=debug)

failed tests:
JIT\\jit64\\hfa\\main\\testG\\hfa_nd0G_d\\hfa_nd0G_d.cmd
JIT\\jit64\\hfa\\main\\testG\\hfa_nd1G_d\\hfa_nd1G_d.cmd

CoreCLR windows x86 Checked gcstress0xc @ Windows.10.Amd64.Open

Error message
~~~
Assert failure(PID 2932 [0x00000b74], Thread: 3164 [0x0c5c]): CheckInstrBytePattern(base[offset ] & 0xFD, 0x81, base[offset]) && CheckInstrBytePattern(base[offset+1] & 0xC0, 0xC0, base[offset+1])

CORECLR! SKIP_ARITH_REG + 0xBA (0x70aa1a7a)
CORECLR! SKIP_ALLOC_FRAME + 0x165 (0x70aa1965)
CORECLR! UnwindEspFrameProlog + 0x148 (0x70aa27c8)
CORECLR! UnwindStackFrame + 0x2E8 (0x70aa2b48)
CORECLR! DoGcStress + 0x291 (0x70c79d02)
CORECLR! OnGcCoverageInterrupt + 0x160 (0x70c7a2e6)
CORECLR! IsGcMarker + 0x93 (0x70ab6c7e)
CORECLR! CLRVectoredExceptionHandlerShim + 0xC5 (0x70aab755)
NTDLL! WinSqmEventWrite + 0x1C907 (0x77b90317)
NTDLL! RtlUnwind + 0x1BA (0x77b503ea)
    File: F:\workspace\_work\1\s\src\coreclr\src\vm\eetwain.cpp Line: 3005
    Image: C:\h\w\966F0885\p\CoreRun.exe


Return code:      1
Raw output file:      C:\h\w\966F0885\w\B069099D\e\JIT\jit64\Reports\JIT.jit64\hfa\main\testG\hfa_nd1G_d\hfa_nd1G_d.output.txt
Raw output:
BEGIN EXECUTION
 "C:\h\w\966F0885\p\corerun.exe" hfa_nd1G_d.dll 
[Add01 HFA 01]PASSED
[Add01 HFA 02]PASSED
[Add01 HFA 03]PASSED
[Add01 HFA 05]PASSED
[Add01 HFA 08]PASSED
[Add01 HFA 11]PASSED
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\966F0885\p
> C:\h\w\966F0885\w\B069099D\e\JIT\jit64\hfa\main\testG\hfa_nd1G_d\hfa_nd1G_d.cmd
Expected: True
Actual:   False


Stack trace
   at JIT_jit64._hfa_main_testG_hfa_nd1G_d_hfa_nd1G_d_._hfa_main_testG_hfa_nd1G_d_hfa_nd1G_d_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\windows.x86.Checked\TestWrappers\JIT.jit64\JIT.jit64.XUnitWrapper.cs:line 9783
~~~

**Comment by BruceForstall** [comment]
@janvorli I haven't seen this GCStress failure mode before.

cc @AndyAyersMS

**Comment by janvorli** [comment]
It looks like we are generating a new instruction sequence that the x86 unwinding doesn't understand. Since the call stack mentions SKIP_ALLOC_FRAME as the caller of SKIP_ARITH_REG, it seems it is related to the stack probing loop.

**Comment by AndyAyersMS** [comment]
Right. Wonder if it is #44664?

**Comment by BruceForstall** [comment]
@echesakovMSFT

**Comment by echesakovMSFT** [comment]
@janvorli Can you please tell if I got this right - there is no unwinding codes on x86, so the VM parses prolog instructions manually. This is what happens in `SKIP_ALLOC_FRAME` that tries to predict what probing sequence the JIT would use given a stack frame size. Since we changed the sequence we need to update that function as well?

**Comment by janvorli** [comment]
@echesakovMSFT yes, that's right.

**Comment by echesakovMSFT** [comment]
I have started working on a solution to the problem. Given that the case is quite *rare* (esp-based frame with the size greater than 1 page size) and we haven't had the test coverage for that until recently (#44664 - when I switched to always using a helper when stack probing is needed) I would like to spend some time on constructing more test cases like these to have better test coverage. In the meantime, I propose to temporary disable the failing tests to reduce the noise in the next GCStress runs.
